consent/self-service-portal/cdr_bank_client.go

```
@@ -1,39 +1,137 @@
 package main
 
-type CDRBankClient struct{}
-
-func (c *CDRBankClient) GetInternalAccounts(subject string) (InternalAccounts, error) {
-	if subject == "user" {
-		return InternalAccounts{
-				Accounts: []InternalAccount{
-					{
-						ID:   "96534987",
-						Name: "Digital banking account",
-					},
-					{
-						ID:   "1000001",
-						Name: "Savings",
-					},
-					{
-						ID:   "1000002",
-						Name: "Savings 2",
-					},
-				},
-			},
-			nil
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/cloudentity/openbanking-quickstart/openbanking/cdr/banking/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+type CDRBankClient struct {
+	httpClient       *http.Client
+	bankClientConfig BankClientConfig
+	cc               clientcredentials.Config
+	baseURL          string
+}
+
+func NewCDRBankClient(config Config) BankClient {
+	var (
+		pool  *x509.CertPool
+		cert  tls.Certificate
+		data  []byte
+		certs []tls.Certificate
+		err   error
+	)
+
+	if pool, err = x509.SystemCertPool(); err != nil {
+		logrus.Fatalf("failed to read system root CAs %v", err)
 	}
 
-	return InternalAccounts{
-			Accounts: []InternalAccount{
-				{
-					ID:   "96565987",
-					Name: "Credit",
-				},
-				{
-					ID:   "1122334455",
-					Name: "Savings",
+	if data, err = os.ReadFile(config.RootCA); err != nil {
+		logrus.Fatalf("failed to read http client root ca: %v", err)
```

return err instead of logging